src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHash128.cs

```diff
@@ -211,17 +211,10 @@ public UInt128 GetCurrentHashAsUInt128()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static void WriteBigEndian128(in Hash128 hash, Span<byte> destination)
         {
-            ulong low = hash.Low64;
-            ulong high = hash.High64;
-            if (BitConverter.IsLittleEndian)
-            {
-                low = BinaryPrimitives.ReverseEndianness(low);
-                high = BinaryPrimitives.ReverseEndianness(high);
-            }
-
-            ref byte dest0 = ref MemoryMarshal.GetReference(destination);
-            Unsafe.WriteUnaligned(ref dest0, high);
-            Unsafe.WriteUnaligned(ref Unsafe.AddByteOffset(ref dest0, new IntPtr(sizeof(ulong))), low);
+            Span<byte> destHi = destination.Slice(0, 8);
+            Span<byte> destLo = destination.Slice(8, 8);
+            BinaryPrimitives.WriteUInt64BigEndian(destLo, hash.Low64);
+            BinaryPrimitives.WriteUInt64BigEndian(destHi, hash.High64);
```

Comment:
IIRC, we've had issues in the past with XxHash with inlining not inlining things we expected it to inline because it was hitting inlining budgets. It'd be good to confirm this has no effect on the generated code / throughput.